extension/src/NABfunctions.ts

```diff
@@ -151,9 +154,28 @@ export async function findTranslatedTexts() {
                 throw new Error('This line does not contain any translated property or label.');
             }
             const textToSearchFor = selectedMlObject[0].xliffId();
-            let fileFilter = '';
-            if (Settings.getConfigSettings()[Setting.SearchOnlyXlfFiles] === true) { fileFilter = '*.xlf'; }
-            await VSCodeFunctions.findTextInFiles(textToSearchFor, false, fileFilter);
+            let langFiles = (await WorkspaceFunctions.getLangXlfFiles(vscode.window.activeTextEditor.document.uri));
+            let doSearch = true;
+            if (langFiles.length === 1) {
```

Comment:
Couldn't `doSearch` be replaced by `if (langFiles.length ...) { ... } else { ... }` ?

Reply:
there too many nested conditions until we know the final result of doSearch so I couldn't solve it like that... could you?

Reply:
Just re-read the README and realized you want to fall back on `findTextInFiles`. I guess you could test with the `targetResult`. Then you could av the two end operations in the same condition as well. Things could be done, it doesn't necessarily be in this PR :)

Reply:
Yes, it's the fallback that made things more complicated. But I think the new refactored approach made things easier, right?

Reply:
Well never mind :+1: